src/ChangelogGenerator.php

```diff
@@ -0,0 +1,230 @@
+                 }
+
+                 $data .= "\n";
+             }
+         }
+
+         return $data;
+     }
+
+     /**
+      * Get all issues from release tags.
+      *
+      * @param mixed $startDate [description]
+      *
+      * @return array [description]
+      *
+      * @throws Exception
+      */
+     private function collectReleaseIssues($startDate = null)
+     {
+         $releases = $this->repository->getReleases();
+
+         if (empty($releases)) {
+             throw new Exception('No releases found for this repository');
+         }
+
+         do {
+             $currentRelease = current($releases);
+
+            if ($startDate &&
+                date_diff(new DateTime($currentRelease->published_at), new DateTime($startDate))->days <= 0
+            ) {
+                continue;
+            }
+
+             $lastRelease = next($releases);
+             $lastReleaseDate = $lastRelease ? $lastRelease->published_at : null;
+             prev($releases);
+
+             $currentRelease->issues = $this->collectIssues($lastReleaseDate);
+
+         } while (next($releases));
+
+         return $releases;
+     }
+
+     /**
+      * Get all issues from release date.
+      *
+      * @param mixed $lastReleaseDate [description]
+      *
+      * @return array [description]
+      */
+     private function collectIssues($lastReleaseDate)
+     {
+         if (!$this->currentIssues) {
+             $this->currentIssues = $this->repository->getIssues(['state' => 'closed']);
+         }
+
+         $issues = [];
+
+         foreach ($this->currentIssues as $x => $issue) {
+             if (new \DateTime($issue->closed_at) > new \DateTime($lastReleaseDate) || $lastReleaseDate == null) {
+                 unset($this->currentIssues[$x]);
+
+                 $type = $this->getTypeFromLabels($issue->labels);
+
+                 if (!$type && isset($issue->pull_request)) {
+                     $type = $this::LABEL_TYPE_PR;
+                 }
+
+                 if ($type) {
+                     $events = $this->repository->getIssueEvents($issue->number);
+                     $isMerged = false;
+
+                     foreach ($events as $event) {
+                         if (in_array($event->event, self::$supportedEvents) && !empty($event->commit_id)) {
+                             $isMerged = true;
+                             break;
+                         }
+                     }
+
+                     if ($isMerged) {
+                         $issues[$type][] = $issue;
+                     }
+                 }
+             }
+         }
+
+         return $issues;
+     }
+
+     /**
+      * Gets the issue type from issue labels.
+      *
+      * @param array $labels [description]
+      *
+      * @return mixed [description]
+      */
+     private function getTypeFromLabels(array $labels)
+     {
+         $type = null;
```

Comment:
Hmm, how so? I believe this is being called from `ChangelogGenerator::collectIssues`.